### ISSUE TITLE
Constructors issues for compilation

### ISSUE THREAD

**Issue by miranha** [overturetool/vdm2c#55]
Using different "types" of constructors (e.g. public, private protected)

Having different constructors in a VDM model, such as:

```
public Controller : RealPort * RealPort * RealPort * RealPort * nat * Role * set of Controller * Supervisor ==> Controller
Controller(sensorTemp, valveOpening, fanSpeed, setPoint, id, role_, slaves_,sup_) == (
	skip;
);
protected Controller : () ==> Controller
Controller() == skip; 
```

This will cause an error for the c code compilation, since we get the error "static declaration of '_Z10ControllerEV' follows non-static declaration", since that empty arguments constructor is generated as "static", because it is "protected" in the VDM model

**Comment by peterwvj** [comment]
Thanks for reporting. As far as I recall this involves changing something in the templates. @bandurvp are you on to this? If it involves changing some of the transformations then I'm happy to help.

**Comment by bandurvp** [comment]
Yes I will take a look at this, but I don't think I can do it before I return from the meeting in Paris next week.  Or while I'm there if I get the time.

**Comment by peterwvj** [comment]
Thanks, Victor. That's fine, I think Miran plans to work on the case study some time after Paris.

**Comment by miranha** [comment]
Victor, we can have a talk in Paris about it. I can also add it. I think the problem in this case is the order of the generation of the different functions.

**Comment by bandurvp** [comment]
Miran, the test I have for constructors contains exactly the same structure as what you indicate above:  I have a default constructor that does not take any parameters but which does have a custom body.  The custom body gets implemented and inserted in the body of the generated default constructor, so there is no interference.  Maybe we should sit down and compare output.  If you generate from the branch vpb/development, do you still get this error?

**Comment by bandurvp** [comment]
I take that back, I totally overlooked the "protected" part.  Please stand by.

**Comment by bandurvp** [comment]
Fixed in commit c8c4b06a7378e41607e59303011f69c33104cd83.
